### DISCUSSION TITLE
Why are swift_projectBox and swift_release commented out?

### DISCUSSION THREAD

**Discussion by stephencelis** [Azoy/Echo#59] [Q&A]
Following discussion here https://github.com/Azoy/Echo/discussions/40#discussioncomment-657168, and the linked reimplementation of `Mirror`, I found that it relied on `swift_projectBox` and `swift_release`, which are both currently commented out on the `main` branch. Is there a reason these functions have been commented out? Is there a way to reimplement `Mirror` without them using other tools?

**Comment by NSExceptional** [comment]
I suspect there are other ways to allow the compiler to trigger a release for you, but there's definitely a way to pull things out of an `AnyExistentialContainer` without `swift_projectBox` (assuming that's what it's for?)

Looks like Azoy has been pretty busy since joining Apple. I can try to help if there's something you're trying to do in particular?

**Reply by stephencelis** [reply]
@NSExceptional My questions came about when I was looking to add runtime enum extraction functionality to [swift-case-paths](https://github.com/pointfreeco/swift-case-paths). We were previously traversing a mirror, which was reliable but about 1,000x slower than a manual `if case let`. A couple of us were able to work out most of the runtime details (seen in [this file](https://github.com/pointfreeco/swift-case-paths/blob/f73f4939ff6adb1826b61d56344245bcac4a73de/Sources/CasePaths/EnumReflection.swift)), and now the implementation is only about 2-5x slower than manual `if case let`.

I keep dreaming for enum key paths to make it into the language though 😅
